core/src/main/java/org/kiwix/kiwixmobile/core/history/HistoryAdapter.kt

```diff
@@ -44,16 +44,14 @@ class HistoryAdapter(
 
   override fun onBindViewHolder(holder: ViewHolder, position: Int) {
     if (holder is HistoryItemViewHolder) {
-      val historyItem = historyList[position] as HistoryItem
-      holder.bind(historyItem)
+      holder.bind(historyList[position] as HistoryItem)
     } else if (holder is HistoryCategoryItemViewHolder) {
-      val date = (historyList[position] as DateItem).dateString
-      holder.bind(date)
+      holder.bind(historyList[position] as DateItem)
     }
   }
 
   override fun getItemViewType(position: Int): Int =
-    if (historyList[position] is DateItem) 0 else TYPE_ITEM
+    if (historyList[position] is DateItem) DateItem.TYPE_CATEGORY else DateItem.TYPE_ITEM
```

Comment:
No, you misunderstood me. This method is asking you to `getItemViewType`

We check if the currentItem `historyList[position]` is a `DateItem` and then we should access `DateItem.VIEW_TYPE`
else the item is a HistoryItem (this is inferred which is not great design but we are letting it slide) so it should be HistoryItem.VIEW_TYPE.

`Category` should be purged from every name in this set of classes, the Date Header is not a Category so the name is misleading.

Reply:
like this then?

```

override fun getItemViewType(position: Int): Int =
    if (historyList[position] is DateItem) DateItem.VIEW_TYPE else HistoryItem.VIEW_TYPE

```

and then 

```
  data class DateItem(
    ...
    companion object {
      const val VIEW_TYPE = 0
    }
  }
```
and
```
  data class HistoryItem constructor(
    ...
    companion object {
      const val VIEW_TYPE = 1
    }
  }
```

Reply:
Yup, looks good, you could even forgo the companion object and declare a file level `const`